File: 정민선/프로그래머스/숫자 문자열과 영단어.py
def solution(s):
    answer = ''
    text = list(s)
    
    strNumber = {
        'zero': '0', 'one': '1', 'two': '2', 'three': '3', 'four': '4', 
        'five': '5', 'six': '6', 'seven': '7', 'eight': '8', 'nine': '9'
    }
    
    temp = ''
    for i in range(len(text)):
        try:
            n = int(text[i])
            answer += text[i]
        except:
            temp += text[i]
            
            for sn in strNumber:
                if sn == temp:
                    answer += strNumber[temp]
                    temp = ''

    return int(answer)

File: 이진영/프로그래머스/숫자 문자열과 영단어.js
function solution(s) {
    const words = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine'];
    const reg = new RegExp("[0-9]|"+ words.join("|"), "g");
    return +s.match(reg).map(i => isNaN(i) ? words.indexOf(i) : i).join("")
}